src/datepicker/datepicker.js

```diff
@@ -118,6 +120,25 @@ export default class Datepicker extends React.Component<
 
   handleInputChange = (event: SyntheticInputEvent<HTMLInputElement>) => {
     const inputValue = event.currentTarget.value;
+    const inputHasNumbers = new RegExp('[1-9]').test(inputValue);
+    const inputHasLetters = new RegExp('[a-z]', 'i').test(inputValue);
+
+    // without letters and numbers, chances are, the user cleared the input
+    // and only the mask is present
+    if (!inputHasNumbers && !inputHasLetters) {
+      if (this.props.range) {
+        this.props.onChange &&
+          this.props.onChange({
+            // $FlowFixMe
+            date: [null, null],
```

Comment:
if it's an array, items in it must be defined: https://github.com/uber/baseweb/blob/master/src/datepicker/types.js#L19